src/serial.rs

```diff
@@ -277,12 +322,12 @@ impl<W: Write> Serial<W> {
                     if self.in_buffer.len() < LOOP_SIZE {
                         self.in_buffer.push_back(value);
                         self.set_lsr_rda_bit();
-                        self.received_data_interrupt()?;
+                        self.received_data_interrupt().map_err(Error::Trigger)?;
```

Comment:
I agree with @andreeaflorescu that these are a pain but they're unavoidable. The `From` impl would generate a conflict on the `io::Error` branch (it's basically `impl<T> From<T> for T` and rustc doesn't approve). So let's leave it this way